gettsim/transfers/rente.py

```diff
@@ -274,25 +300,419 @@ def ges_rente_zugangsfaktor(
     return out
 
 
-def ges_rente_regelaltersgrenze(geburtsjahr: int, ges_rente_params: dict) -> float:
-    """Calculates the age, at which a worker is eligible to claim his full pension.
+def ges_rente_regelaltersgrenze(
+    geburtsjahr: int, ges_rente_vorraus_regelrente: bool, ges_rente_params: dict
+) -> float:
+    """Calculates the age, at which a person is eligible to claim the regular pension.
+        Normal retirement age (NRA). This pension cannot be claimed earlier than at the
+        NRA, ie it does not serve as reference for calculating deductions. However, it
+        serves as reference for calculating gains in the Zugangsfakor in case
+        of later retirement.
+
+    Parameters
+    ----------
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+    ges_rente_vorraus_regelrente
+        See :func: ``ges_rente_vorraus_regelrente``.
+
+    Returns
+    -------
+    """
+    if ges_rente_vorraus_regelrente:
+        out = piecewise_polynomial(
+            x=geburtsjahr,
+            thresholds=ges_rente_params["regelaltersgrenze"]["thresholds"],
+            rates=ges_rente_params["regelaltersgrenze"]["rates"],
+            intercepts_at_lower_thresholds=ges_rente_params["regelaltersgrenze"][
+                "intercepts_at_lower_thresholds"
+            ],
+        )
+    else:
+        out = 9000  # arbitrary high number to represent non-feasible retirement age
+    return out
+
+
+def ges_rente_grenze_langjährig_frau(
+    geburtsjahr: int,
+    geburtsmonat: int,
+    ges_rente_params: dict,
+    ges_rente_vorraussetz_frauen: bool,
+    ges_rente_vorraussetz_langjährig: bool,
+) -> float:
+    """Calculates the age, at which a longterm insured person or women is eligible to
```

Comment:
I don't understand the difference between `ges_rente_grenz_voll_altersrente` and `ges_rente_grenze_langjährig_frau` and why we need both in this function.

Reply:
ges_rente_grenz_volll_altersrente is defining the age-threshold from which the person is able to claim the full pension (zugangsfaktor=1). 
ges_rente_grenze_lanjährig_frau is the threshold from which the person is able to claim either the pension for women or the pension for long-term insured (importantly, it does not include the rente mit 63)- it can coincide with the former but does not have to. This threshold becomes important when the zugangsfaktor is smaller than 1, because then the deduction (zugangsfaktor) is calculated relative to this threshold.

Reply:
Thanks! A couple of notes:

- Please don't overload the function with two different things, even if they happened to be the similar in the past. So there should be two functions `ges_rente_grenze_langjährig` and `ges_rente_grenze_frauen` -- good thing going forward is, for example, that we can ignore the second function.
- For the last line taking the minimum of both, add another function. Is there a technical term for this age, i.e., when calculating the `zugangsfaktor`?
- Use German identifiers also inside the function instead of translations like `pension_for_women`, `pension_for_long` (plus, not clear we are talking about age there and not the pension itself)
- Please don't abbreviate identifiers written out in full elsewhere, that makes searching the codebase hard (`grenz_` in `ges_rente_grenz_voll_altersrente`)

Reply:
> Thanks! A couple of notes:
> 
>     * Please don't overload the function with two different things, even if they happened to be the similar in the past. So there should be two functions `ges_rente_grenze_langjährig` and `ges_rente_grenze_frauen` -- good thing going forward is, for example, that we can ignore the second function.
> 
now in two separate functions

>     * For the last line taking the minimum of both, add another function. Is there a technical term for this age, i.e., when calculating the `zugangsfaktor`?
> 
It is now done in the function `referenz_alter_abschlag`. Hope this this makes it more transparent. I am not aware of an  official technical term for this.

>     * Use German identifiers also inside the function instead of translations like `pension_for_women`, `pension_for_long` (plus, not clear we are talking about age there and not the pension itself)
> 
I got rid of these terms.

>     * Please don't abbreviate identifiers written out in full elsewhere, that makes searching the codebase hard (`grenz_` in `ges_rente_grenz_voll_altersrente`)
I consolidated the terms

Reply:
Excellent! I think I saw "pension for women" in a docstring somewhere, please also check those at some point.

Reply:
changed to german terms